src/ray/raylet/node_manager.cc

```diff
@@ -2536,6 +2600,15 @@ void NodeManager::FinishAssignedActorTask(Worker &worker, const Task &task) {
       worker.MarkDetachedActor();
     }
 
+    if (RayConfig::instance().gcs_service_enabled()) {
+      // The gcs server is responsible for notifying other nodes of the changes of actor
+      // status. By the way, the `FinishAssignedActorTask` will never be triggered if
+      // `new_scheduler_enabled_` is true.
```

Comment:
This comment can possibly be rephrased to:

```
      // Gcs server is responsible for notifying other nodes of the changes of actor
      // status, and thus raylet doesn't need to handle this anymore.
      //  And if `new_scheduler_enabled_` is true, this function `FinishAssignedActorTask`
      // will not be called because raylet is not aware of the actual task when receiving
      // a worker lease request.
```